tests/federation_room_join_partial_state_test.go

```diff
@@ -3718,6 +3718,10 @@ func TestPartialStateJoin(t *testing.T) {
 
 			// Cleanup.
 			psjResult.FinishStateRequest()
+			// Dirty hack to allow the homeserver under test to finish making requests to the
+			// Complement homeserver as part of syncing the full state.
+			psjResult.AwaitStateIdsRequest(t)
+			time.Sleep(time.Second / 2)
```

Comment:
This is incredibly hacky. Alternative suggestions welcome.

Reply:
Top of my head: we could make the homeserver under test take an action in the room that requires knowledge of the full state, and wait for complement's HS to see it. (Is there an example of a federation API that blocks on full state?)

(Personally I can live with the hack though)

Reply:
I think the difficulty is that the homeserver under test leaves the room independently of, and before the sync process finishes. Any request we might try would be met with a not-in-the-room response even though the sync process may still be running.